src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -292,92 +196,186 @@ internal sealed class MeterProviderSdk : MeterProvider
         }
         else
         {
-            this.listener.InstrumentPublished = (instrument, listener) =>
-            {
-                bool enabledMeasurements = false;
+            // Everything double
+            this.listener.SetMeasurementEventCallback<double>(this.MeasurementRecordedDoubleSingleStream);
+            this.listener.SetMeasurementEventCallback<float>((instrument, value, tags, state) => this.MeasurementRecordedDoubleSingleStream(instrument, value, tags, state));
+
+            // Everything long
+            this.listener.SetMeasurementEventCallback<long>(this.MeasurementRecordedLongSingleStream);
+            this.listener.SetMeasurementEventCallback<int>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+            this.listener.SetMeasurementEventCallback<short>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+            this.listener.SetMeasurementEventCallback<byte>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+
+            this.listener.MeasurementsCompleted = (instrument, state) => this.MeasurementsCompletedSingleStream(instrument, state);
+        }
+
+        this.listener.Start();
+
+        OpenTelemetrySdkEventSource.Log.MeterProviderSdkEvent("MeterProvider built successfully.");
+    }
+
+    internal Resource Resource { get; }
+
+    internal List<object> Instrumentations => this.instrumentations;
+
+    internal MetricReader? Reader => this.reader;
+
+    internal int ViewCount => this.viewConfigs.Count;
+
+    internal object? InstrumentPublished(Instrument instrument, bool listeningIsManagedExternally)
+    {
+        var listenToInstrumentUsingSdkConfiguration = this.shouldListenTo(instrument);
+
+        if (listeningIsManagedExternally && listenToInstrumentUsingSdkConfiguration)
+        {
+            OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(
+                instrument.Name,
+                instrument.Meter.Name,
+                "Instrument belongs to a Meter which has been enabled both externally and via a subscription on the provider. External subscription will be ignored in favor of the provider subscription.",
+                "Programmatic calls adding meters to the SDK (either by calling AddMeter directly or using 'AddInstrumentation' extensions) are always favored over external registrations. When also using external management (typically IMetricsBuilder or IMetricsListener) remove programmatic calls to the SDK to allow registrations to be added and removed dynamically.");
```

Comment:
Other than just instrumentation extension methods, there might be other methods too that might call `AddMeter`. Maybe we could point out that programmatic calls include any kind of indirect calls to `AddMeter`?

```suggestion
                "Programmatic calls adding meters to the SDK (either by calling AddMeter directly or using 'AddInstrumentation' extensions or any indirect calls to `AddMeter`) are always favored over external registrations. When also using external management (typically IMetricsBuilder or IMetricsListener) remove programmatic calls to the SDK to allow registrations to be added and removed dynamically.");
```

Reply:
I tweaked it please resolve this comment if you are good with this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4958/commits/da5554b91da5198f87a1158344de1e6ab5da9cec